Make passthrough a toggle on first person controller

BastiaanOlij opened the pull request:
Right now the properly on the first person controller script just auto starts passthrough.

This change makes it a proper toggle with which to turn on/off passthrough.

@m4gr3d is this the correct way to implement this?

m4gr3d commented on demo/addons/godot-openxr/scenes/first_person_controller_vr.gd:
```diff
@@ -35,8 +48,12 @@ func initialise() -> bool:
+
 		# Start passthrough?
-		_start_passthrough()
+		if enable_passthrough:
+			_start_passthrough()
```
Here you should follow the same pattern as in `set_passthrough(...)` and update `enable_passthrough` with the result from `_start_passthrough`.

BastiaanOlij replied:
Ok, thats a misunderstanding.

These are properties so the values are stored in the TSCN file. After loading Godot will call set_passthrough with the stored value however this is called as part of the loading process when we're not yet ready to call `_start_passthrough`. That is why we call `_start_passthrough` here based on whether the setting was set during load.

While I used to do a syntax here like `set_passthrough(enable_passthrough)`, I've stopped doing this because in Godot 4 the syntax for properties changed and this would result in a never ending loop as setting `enable_passthrough` would call `set_passthrough` again.

m4gr3d commented on demo/addons/godot-openxr/scenes/first_person_controller_vr.gd:
```diff
@@ -74,13 +93,21 @@ func _get_xr_viewport() -> Viewport:
-	if start_passthrough:
-		# make sure our viewports background is transparent
-		_get_xr_viewport().transparent_bg = true
+func is_passthrough_supported() -> bool:
+	return enabled_extensions.find("XR_FB_passthrough") >= 0
```
Is this method being used?

BastiaanOlij replied:
Not yet, it's there for convenience, I'm assuming people who want to support this will want to check if support is available.

m4gr3d commented on demo/addons/godot-openxr/scenes/first_person_controller_vr.gd:
```diff
@@ -74,13 +93,21 @@ func _get_xr_viewport() -> Viewport:
+
+func _start_passthrough() -> bool:
+	# make sure our viewports background is transparent
+	_get_xr_viewport().transparent_bg = true
```
Is there a way to detect the version of Godot being used and print a warning for https://github.com/GodotVR/godot_openxr/issues/139?

BastiaanOlij replied:
We should have access to the version number, I'll need to check.

BastiaanOlij replied:
Added that in with a few other checks that seemed to make sense.

m4gr3d reviewed (changes requested):
Just some minor feedback, but it looks mostly good.

Based on the updated logic, I realize I need to update the `start_passthrough` logic as the current implementation doesn't deal well with calling it several times in a row. I'll provide a PR with the update.

m4gr3d commented:
> Based on the updated logic, I realize I need to update the `start_passthrough` logic as the current implementation doesn't deal well with calling it several times in a row. I'll provide a PR with the update.

Updated the logic in https://github.com/GodotVR/godot_openxr/pull/142

m4gr3d commented:
@BastiaanOlij Were you able to update this PR following #142?

BastiaanOlij commented:
@m4gr3d I rebased this project and added a few extra things into it, including being able to switch pass through with a button in the UI that is shown. Works like a charm on my Quest 1 though the demo scene isn't very suitable for pass through, the ground plane definitely gets in the way :)
